src/liblzma/check/crc32_loongarch.h

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: 0BSD
+
+///////////////////////////////////////////////////////////////////////////////
+//
+/// \file       crc32_loongarch.h
+/// \brief      CRC32 calculation with LoongArch optimization
+//
+//  Authors:    Xi Ruoyao
+//
+///////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef LZMA_CRC32_LOONGARCH_H
+#define LZMA_CRC32_LOONGARCH_H
+
+#include <larchintrin.h>
+
+static uint32_t
+crc32_arch_optimized(const uint8_t *buf, size_t size, uint32_t crc_unsigned)
+{
+	int32_t crc = (int32_t)~crc_unsigned;
+
+	// Align the input buffer because this was shown to be
+	// significantly faster than unaligned accesses.
+	const size_t align_amount = my_min(size,
+			(8 - (uintptr_t)buf) & (8 - 1));
```

Comment:
`(0 - (uintptr_t)buf)` does the same thing and should map to a negation instruction. Possibly a compiler will do the same from the current code too but using a zero makes it explicit. A similar change was done in 2337f7021c860b026e3e849e60a9ae8d09ec0ea0. Using `0U` instead of `0` isn't important here as `uintptr_t` will promote the math to unsigned anyway so it's just what one prefers.

Some would write just `(-(uintptr_t)buf)` but that can cause warnings due to sign conversion, thus I prefer writing it as a subtraction from zero.